### ISSUE TITLE
Implement a review backend

### ISSUE THREAD

**Issue by razzeee** [flathub/backend#54]
Basically mirroring [odrs](odrs.gnome.org/), I think.

https://github.com/flathub/linux-store-frontend/pull/311

**Comment by bilelmoussaoui** [comment]
Odrs contains reviews submitted by people using the same software but packaged differently. A bunch of issues could be caused because of that.

Showing those reviews doesn't make sense to me.

**Comment by barthalion** [comment]
I don't see a field designating flatpak packages, so I agree, it doesn't make much sense as it is.

**Comment by barthalion** [comment]
In fact, none of the reviews in the example call for Firefox are actually about Flathub, as we're using lowercase `org.mozilla.firefox` as agreed with Mozilla, while all entries are either Snap, distro (`firefox.desktop`) or Fedora (`org.mozilla.Firefox`).

**Comment by razzeee** [comment]
yeah, I've briefly talked about us not being able to separate the different packaging providers in the flatpak matrix channel. 

I guess there are tree ways forward:
1. Accept that we can't do it
1. Update odrs to be able to hold this data and make sure that distros will send this at some point -> then implement it based on that.
1. Have our own non odrs backend (which has a bunch of other implications)
1. Mix them up and don't care about packaging causing reviews (you could argue, that most reviews shouldn't be about packaging)

**Comment by razzeee** [comment]
Thinking about this again, Gnome Software seems to somehow differentiate reviews between rpm, flatpak (fedora) and flatpak (flathub) so I guess it's possible somehow?

**Comment by razzeee** [comment]
This would need to also add the reviews/ratings from the linked provides->id
https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-provides

**Comment by barthalion** [comment]
> Thinking about this again, Gnome Software seems to somehow differentiate reviews between rpm, flatpak (fedora) and flatpak (flathub) so I guess it's possible somehow?

Do you have screenshots that show this? GS is able to change the repo to install from, but it seems to me that reviews stay the same.

**Comment by razzeee** [comment]
Not sure if I remember right and this might be totally invalid:

![image](https://user-images.githubusercontent.com/5943908/140616094-524600f6-af5c-4ea1-9784-d3578442b81d.png)

![image](https://user-images.githubusercontent.com/5943908/140616107-235f46a1-5f43-4408-958c-a3142dd96288.png)

![image](https://user-images.githubusercontent.com/5943908/140616144-0d89eaa0-4bfa-4019-829d-030f51a4be3b.png)

I tried to look into that when I reported it, but I don't think the code is easy to follow, it might just be due to the app id and the provides->id accumulating differently as we're switching the metadata file.

**Comment by barthalion** [comment]
Yeah, I think it's something with provided ID but ultimately the same reviews to a degree.

**Comment by razzeee** [comment]
Still feels quiet misleading and shouldn't they load up those reviews too? Not just in the sum?

**Comment by razzeee** [comment]
I still think ratings would be cool, even if we just show the stars and nothing else.

Then again, I think most people that are looking for apps, will not care about how it's packaged (they might, but I think that assumption is due to us working on this exactly) and the reviews are mostly not mentioning how it's packaged.
